GetandTake/Controller/ProductController.cs

```diff
@@ -0,0 +1,29 @@
+﻿using GetandTake.Models.DTOs.ListDTO;
+using GetandTake.Services.Abstract;
+using Microsoft.AspNetCore.Mvc;
+
+namespace GetandTake.Controller;
+
+[Route("api/[controller]")]
+[ApiController]
+public class ProductController : ControllerBase
+{
+    private readonly IProductService productService;
+    public ProductController(IProductService _productService)
+    {
+        productService = _productService;
+    }
+
+    [HttpGet("productlist")]
+    public async Task<List<ProductsDTO>> ProductList()
+    {
+        var productList = await productService.GetAllAsync();
+        return productList;
+
+    }
+    [HttpGet("getproductbyid")]
```

Comment:
Same case with URL, the only case is just to add URL postfix with type constraints, like:
```suggestion
    [HttpGet("id/{id:int}")]
```